stratum/hal/lib/tdi/tdi_sde_table_entry.cc

```diff
@@ -65,11 +65,13 @@ ::util::Status TdiSdeWrapper::InsertTableEntry(
    */
 
   const auto flags = ::tdi::Flags(0);
-  RETURN_IF_TDI_ERROR(table->entryAdd(
+  tdi_status_t status = table->entryAdd(
```

Comment:
Are you sure that this is the correct behavior in all cases? What if the caller attempts to add an entry with the same key but different data?

On the other hand, if the caller is expected to have verified the non-existence of the entry before issuing the Add, then failure could reasonably be considered an internal error.

If the answer is "it depends", then perhaps this method should return ::util::error::Code::ALREADY_EXISTS and allow the caller to make the decision.

(Apologies if I've already asked this. This change looks vaguely familiar.)

Reply:
Yes, this is correct behavior. If, table key is same and data/action associated with it has changed, then user is expected to use ::p4::v1::Update::Type as MODIFY and  ModifyTableEntry SDE wrapper API will be invoked. I am not sure, if there is support for this in the backend. I will check with SDE team and if yes, then we can print the message accordingly.